packages/stores/src/ui-config/trade-token-in-config.ts

```diff
@@ -0,0 +1,327 @@
+import {
+  AmountConfig,
+  IFeeConfig,
+  InsufficientAmountError,
+} from "@keplr-wallet/hooks";
+import { ChainGetter, IQueriesStore } from "@keplr-wallet/stores";
+import { AppCurrency } from "@keplr-wallet/types";
+import {
+  CoinPretty,
+  Dec,
+  DecUtils,
+  Int,
+  IntPretty,
+  RatePretty,
+} from "@keplr-wallet/unit";
+import {
+  OptimizedRoutes,
+  Pool,
+  RoutePathWithAmount,
+} from "@osmosis-labs/pools";
+import { action, computed, makeObservable, observable, override } from "mobx";
+
+export class TradeTokenInConfig extends AmountConfig {
+  @observable.ref
+  protected _pools: Pool[];
+
+  @observable
+  protected _inCurrencyMinimalDenom: string | undefined = undefined;
+  @observable
+  protected _outCurrencyMinimalDenom: string | undefined = undefined;
+  @observable
+  protected _error: Error | undefined = undefined;
+
+  constructor(
+    chainGetter: ChainGetter,
+    queriesStore: IQueriesStore,
+    initialChainId: string,
+    sender: string,
+    feeConfig: IFeeConfig | undefined,
+    pools: Pool[]
+  ) {
+    super(chainGetter, queriesStore, initialChainId, sender, feeConfig);
+
+    this._pools = pools;
+
+    makeObservable(this);
+  }
+
+  @action
+  setPools(pools: Pool[]) {
```

Comment:
Related to `use-token-in-config`, we should remove these simple `set*` methods and have everything be passed into the constructor instead.

If there's a reason to change these core objects that are in the constructor later in the objects lifecycle, LMK and we can keep them.